pkg/alert/client_test.go

```diff
@@ -0,0 +1,183 @@
+package alert
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/thanos-io/thanos/pkg/discovery/dns"
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+func TestBuildAlertmanagerConfiguration(t *testing.T) {
+	for _, tc := range []struct {
+		address string
+
+		err      bool
+		expected AlertmanagerConfig
+	}{
+		{
+			address: "http://localhost:9093",
+			expected: AlertmanagerConfig{
+				StaticAddresses: []string{"localhost:9093"},
+				Scheme:          "http",
+			},
+		},
+		{
+			address: "https://am.example.com",
+			expected: AlertmanagerConfig{
+				StaticAddresses: []string{"am.example.com"},
+				Scheme:          "https",
+			},
+		},
+		{
+			address: "dns+http://localhost:9093",
+			expected: AlertmanagerConfig{
+				StaticAddresses: []string{"dns+localhost:9093"},
+				Scheme:          "http",
+			},
+		},
+		{
+			address: "dnssrv+http://localhost",
+			expected: AlertmanagerConfig{
+				StaticAddresses: []string{"dnssrv+localhost"},
+				Scheme:          "http",
+			},
+		},
+		{
+			address: "ssh+http://localhost",
+			expected: AlertmanagerConfig{
+				StaticAddresses: []string{"localhost"},
+				Scheme:          "ssh+http",
+			},
+		},
+		{
+			address: "dns+https://localhost/path/prefix/",
```

Comment:
Currently, we did not even support paths, so that's nice :+1:  worth to mention in CHANGELOG I think.